src/condor_utils/manifest.cpp

```diff
@@ -198,6 +198,48 @@ deleteFilesStoredAt(
 	std::filesystem::path pathToManifestFile(manifestFileName);
 	std::filesystem::path fileName = pathToManifestFile.filename();
 
+
+	std::string cdmf;
+	param( cdmf, "CHECKPOINT_DESTINATION_MAPFILE" );
+
+	MapFile mf;
+	int rv = mf.ParseCanonicalizationFile( cdmf.c_str(), true, true, true );
+	if( rv < 0 ) {
+		formatstr( error,
+			"Failed to parse checkpoint destination map file (%s), aborting",
+			cdmf.c_str()
+		);
+		return false;
+	}
+
+	std::string argl;
+	if( mf.GetCanonicalization( "*", checkpointDestination.c_str(), argl ) != 0 ) {
+		formatstr( error,
+		    "Failed to find checkpoint destination %s in map file, aborting",
+		    checkpointDestination.c_str()
+		);
+		return false;
+	}
+
+	StringList sl( argl );
+	sl.rewind();
```

Comment:
Can we replace the `StringList` with a `StringTokenIterator` here?

Reply:
Yes.